src/backend/access/appendonly/aosegfiles.c

```diff
@@ -141,7 +142,9 @@ InsertInitialSegnoEntry(Relation parentrel, int segno)
 	if (!HeapTupleIsValid(pg_aoseg_tuple))
 		elog(ERROR, "failed to build AO file segment tuple");
 
-	frozen_heap_insert(pg_aoseg_rel, pg_aoseg_tuple);
+	simple_heap_insert(pg_aoseg_rel, pg_aoseg_tuple);
+	SIMPLE_FAULT_INJECTOR("insert_aoseg_before_freeze");
+	heap_freeze_tuple_wal_logged(pg_aoseg_rel, pg_aoseg_tuple);
```

Comment:
Where is the same for InsertInitialAOCSFileSegInfo() like in original commit?

Reply:
The original commit does not contain an fault injector for InsertInitialAOCSFileSegInfo.
In addition, in 6X, non-frozen lines are inserted there.